testing/internal/e2e/tests/static/target_address_test.go

```diff
@@ -0,0 +1,154 @@
+package static_test
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/boundary/testing/internal/e2e"
+	"github.com/hashicorp/boundary/testing/internal/e2e/boundary"
+	"github.com/stretchr/testify/require"
+)
+
+// TestTargetAddress uses the boundary cli to create a number of supporting objects
+// to connect to a target. It then attempts to connect to that target using a host source
+// and verifies that the connection was successful. Afterwards, we validate that an address
+// cannot be assigned directly to the target that has an existing host source association.
+// The host sources are removed from the same target and a network address is assigned
+// directly to the target. Then it attempts to connect to that target using
+// the direct network address association. Afterwards the test validates that a host source
+// cannot be added or set to a target with an existing network address assigned to it.
+// Finally, the target network address is removed and the host source is added back to the
+// target to attempt a final connection attempt.
+func TestTargetAddress(t *testing.T) {
```

Comment:
We may want to update the name of this test to be more specific, given that we know we have at least 3 tests already defined.

Reply:
Yea, I was having a hard time naming this one, maybe one the following names could be a good replacement:
- TestTargetAddressHostSourceRelationship()
- TestTargetAddressToHostSourceMutuallyExclusiveRelationship()
- TestTargetMutuallyExclusiveRelationship()

I wasn't sure what to pick because I didn't want the function name to feel like I was reading out a sentence, which is why I added a detailed comment to the test.

Reply:
There were three tasks identified on the spreadsheet. Is that still the plan? Does this test cover the first and third items or am I mistaken? Does it make sense to keep these tests separate to help with readability?

Something like...
```
TestCliTargetAddressCreateAndUpdate??
TestCliTargetAddressToHostSource
TestCliTargetHostSourceToAddress
```

Reply:
I just did this - let me know what you think 😃